docs/src/tutorials/discrete_stochastic_example.md

```diff
@@ -200,6 +211,91 @@ to the `DiscreteProblem` and solve it, we would simply do:
+advantageous to use a different internal representation for the jump collection.
+For such systems it is recommended to use `DirectFW()`, which should offer
+better performance than `Direct()`.
+
+## Defining the Jumps Directly: `MassActionJump`
+For systems that can be represented as mass action reactions, a further
+specialization of the jump type is possible that offers improved computational
+performance; `MasssActionJump`. Suppose the system has ``N`` chemical species
+``\{S_1,\dots,S_N\}``. A general mass action reaction has the form
+
+```math
+R_1 S_1 + R_2 S_2 + \dots + R_N S_N \overset{k}{\rightarrow} P_1 S_1 + P_2 S_2 + \dots + P_N S_N
+```
+where the non-negative integers ``(R_1,\dots,R_N)`` denote the *reactant
+stoichiometry* of the reaction, and the non-negative integers
+``(P_1,\dots,P_N)`` the *product stoichiometry*. The *net stoichiometry* is
+the net change in each chemical species from the reaction occurring one time,
+given by ``(P_1-R_1,\dots,P_N-R_N)``.
+
+As an example, consider again the SIR model defined in the `@reaction_network`
+above. The species are then `(s,i,r)`. The first reaction has rate `c1`,
+reactant stoichiometry `(1,1,0)`, product stoichiometry `(0,2,0)`, and net
+stoichiometry `(-1,1,0)`. The second reaction has rate `c2`, reactant
+stoichiometry `(0,1,0)`, product stoichiometry `(0,0,1)`, and net stoichiometry
+`(0,-1,1)`.
+
+We can encode this system as a mass action jump by specifying the rates, reactant
+stoichiometry, and the net stoichiometry as follows:
+```julia
+rates = [0.1/1000, 0.01]    # i.e. [c1,c2]
+reactant_stoich = 
+[
+  [1 => 1, 2 => 1],         # 1*s and 1*i
+  [2 => 1]                  # 1*i
+]
+net_stoich = 
+[
+  [1 => -1, 2 => 1],         # -1*s and 1*i
+  [2 => -1, 3 => 1]         # -1*i and 1*r
+]
+mass_act_jump = MassActionJump(rates, reactant_stoich, net_stoich)
+```
+Just like for `ConstantRateJumps`, to then simulate the system we create
+a `JumpProblem` and call `solve`:
+```julia
+jump_prob = JumpProblem(prob, Direct(), mass_act_jump)
+sol = solve(jump_prob, SSAStepper())
+```
+
+
+## Defining the Jumps Directly: Mixing ConstantJump and MassActionJump
+Suppose we now want to add in to the SIR model another jump that can not be
+represented as a mass action reaction. We can create a new `ConstantRateJump`
+and simulate a hybrid system using both the `MassActionJump` for the two
+previous reactions, and the new `ConstantRateJump`. Let's suppose we want to let
+susceptible people be born with the following jump rate:
+```julia
+birth_rate(u,p,t) = 10.*u[1]/(200. + u[1]) + 10.
+function birth_affect!(integrator)
+  integrator.u[1] += 1
+end
+birth_jump = ConstantRateJump(birth_rate, birth_affect!)
+```
+We can then simulate the hybrid system as
+```julia
+jump_prob = JumpProblem(prob, Direct(), mass_act_jump, birth_jump)
+sol = solve(jump_prob, SSAStepper())
+```
+![gillespie_hybrid_jumps](../assets/gillespie_hybrid_jumps.png)
+
+### Caution about `MassActionJump`s
```

Comment:
this to me reads like a detail that belongs on the solver page.

Reply:
Just to clarify, you mean that the caution about mass action jumps should be on the solver page, or the example of mass action and constant rate jumps?